Code/Framework/AzToolsFramework/Tests/Viewport/ViewportEditorModeTests.cpp

```diff
@@ -152,6 +175,22 @@ namespace UnitTest
         AZStd::array<AZStd::unique_ptr<ViewportEditorModeNotificationsBusHandler>, ViewportEditorModes::NumEditorModes> m_editorModeHandlers;
     };
 
+    // Fixture for testing the integration of viewport editor mode state tracker
+    class ViewportEditorModeTrackerIntegrationTestFixture
+        : public ToolsApplicationFixture
+    {
+    public:
+        void SetUpEditorFixtureImpl() override
+        {
+            m_viewportEditorModeTracker = AZ::Interface<ViewportEditorModeTrackerInterface>::Get();
+            ASSERT_NE(m_viewportEditorModeTracker, nullptr);
```

Comment:
nit: I personally thing these expectations/asserts that are not actually in a test aren't that useful (It can be an AZ_Assert if you think that's a good idea) but if this isn't valid the test will fail anyway